turtle_graphics/turtle_class.js

```diff
@@ -0,0 +1,171 @@
+                    }
+                } else {
+                    for (let i = 1; i <= increase; i++) {
+                        this.arrayOfPOS.push([lastValX, parseInt(lastValY) + i])   
+                    }   
+                }
+                return this;
+                
+            case "left":
+                if (this.arrayOfPOS[0] === this.arrayOfPOS[this.arrayOfPOS.length-1]) {
+                    for (let i = 1; i <= increase; i++) {
+                        this.arrayOfPOS.push([parseInt(this.x) - i, this.y])
+                    }
+                } else {
+                    for (let i = 1; i <= increase; i++) {
+                        this.arrayOfPOS.push([parseInt(lastValX) - i, lastValY])   
+                    }   
+                }
+                return this;
+                
+            case "up":
+                if (this.arrayOfPOS[0] === this.arrayOfPOS[this.arrayOfPOS.length-1]) {
+                    for (let i = 1; i <= increase; i++) {
+                        this.arrayOfPOS.push([this.x, parseInt(this.y) - i])
+                    }
+                } else {
+                    for (let i = 1; i <= increase; i++) {
+                        this.arrayOfPOS.push([lastValX, parseInt(lastValY)- i])   
+                    }   
+                }
+                return this;
+            }
+    }
+
+    right(){
+
+        if (this.direction === "right") {
+            this.direction = "down"
+            return this;
+        } else if (this.direction === "left"){
+            this.direction = "up"
+            return this;
+        }else if(this.direction === "down"){
+            this.direction = "left"
+            return this;
+        }else if(this.direction === "up"){
+            this.direction = "right"
+            return this;
+        }
+    }
+
+    left(){
+
+        if (this.direction === "right") {
+            this.direction = "up"
+            return this;
+        } else if (this.direction === "left"){
+            this.direction = "down"
+            return this;
+        }else if(this.direction === "down"){
+            this.direction = "right"
+            return this;
+        }else if(this.direction === "up"){
+            this.direction = "left"
+            return this;
+        }
+    }
+
+    allPoints(){
+        return this.arrayOfPOS;
+    }
+
+    print(){
+
+        let xArray = [];
+        let yArray = [];
+        let position = ""
+
+        this.arrayOfPOS.forEach(element => {
+            xArray.push(element[0]);
+        });
+        
+        
+        this.arrayOfPOS.forEach(element => {
+            yArray.push(element[1]);
+        });
+        
+        const maxWidth = Math.max(...xArray);
+        const maxHeight = Math.max(...yArray);
+        const minWidth = Math.min(...xArray);
+        const minHeigth = Math.min(...yArray);
+
+            for (let y = minHeigth; y <= maxHeight; y++) {
```

Comment:
nesting for loops 3 levels deep is not suggested. it's fine for this assignment but in real practice it downgrades performance a great deal.